mcp-server/src/analysis.ts

```diff
@@ -0,0 +1,560 @@
+  const standardDeviation = Math.sqrt(variance);
+  const coefficientOfVariation = averageValue > 0 ? standardDeviation / averageValue : 0;
+
+  let volatility: 'low' | 'medium' | 'high' = 'low';
+  if (coefficientOfVariation > 0.3) volatility = 'high';
+  else if (coefficientOfVariation > 0.15) volatility = 'medium';
+  if (volatility === 'high') direction = 'volatile';
+
+  // Calculate momentum
+  const momentum = calculateMomentum(values);
+
+  // Detect growth phases
+  const growthPhases = detectGrowthPhases(values);
+
+  // Simple seasonality detection
+  const hasSeasonality = detectSeasonality(values);
+
+  const timeRangeObj: { start?: string; end?: string } = {};
+  const firstValueEntry = values[0];
+  if (firstValueEntry) timeRangeObj.start = firstValueEntry.date;
+  const lastValueEntry = values[values.length - 1];
+  if (lastValueEntry) timeRangeObj.end = lastValueEntry.date;
+
+  return {
+    dataPoints: values.length,
+    timeRange: timeRangeObj,
+    values: {
+      first: firstValue,
+      last: lastValue,
+      peak: peakValue,
+      lowest: lowestValue,
+      average: averageValue,
+      median: medianValue
+    },
+    trend: {
+      direction,
+      totalGrowth,
+      growthRate: `${growthRate.toFixed(2)}%`,
+      momentum,
+      volatility
+    },
+    patterns: {
+      hasSeasonality,
+      growthPhases
+    }
+  };
+}
+
+/**
+ * Creates an empty trend analysis object with default values.
+ *
+ * @returns An empty TrendAnalysis object.
+ */
+function createEmptyTrendAnalysis(): TrendAnalysis {
+  return {
+    dataPoints: 0,
+    timeRange: {},
+    values: { first: 0, last: 0, peak: 0, lowest: 0, average: 0, median: 0 },
+    trend: {
+      direction: 'stable',
+      totalGrowth: 0,
+      growthRate: '0%',
+      momentum: 'insufficient_data',
+      volatility: 'low'
+    },
+    patterns: { hasSeasonality: false, growthPhases: [] }
+  };
+}
+
+/**
+ * Calculates the momentum of a time-series dataset.
+ *
+ * @param values - Array of time-series data points.
+ * @returns The momentum classification: 'accelerating', 'decelerating', 'stable', or 'insufficient_data'.
+ */
+export function calculateMomentum(values: Array<{date: string; value: number}>): 'accelerating' | 'decelerating' | 'stable' | 'insufficient_data' {
+  if (values.length < 6) return 'insufficient_data';
+
+  const midpoint = Math.floor(values.length / 2);
+  const firstHalf = values.slice(0, midpoint);
+  const secondHalf = values.slice(midpoint);
+
+  const firstHalfGrowth = firstHalf.length > 1
+    ? ((firstHalf[firstHalf.length - 1]!.value - firstHalf[0]!.value) / firstHalf.length)
+    : 0;
+
+  const secondHalfGrowth = secondHalf.length > 1
+    ? ((secondHalf[secondHalf.length - 1]!.value - secondHalf[0]!.value) / secondHalf.length)
+    : 0;
+
+  const momentumDifference = secondHalfGrowth - firstHalfGrowth;
+  const threshold = Math.max(0.1, (firstHalf[0]?.value ?? 0) * 0.01); // Dynamic threshold based on scale
```

Comment:
I think it would be more reasonable for momentum to be related to increments rather than values.